pkg/agent/route/route_linux.go

```diff
@@ -1434,21 +1421,30 @@ func (c *Client) AddLoadBalancer(svcIP net.IP) error {
 // gateway on host.
 func (c *Client) DeleteLoadBalancer(svcIP net.IP) error {
 	svcIPStr := svcIP.String()
-	route, found := c.serviceRoutes.Load(svcIPStr)
-	if !found {
-		klog.V(2).InfoS("Didn't find route for LoadBalancer ingress IP", "ingressIP", svcIPStr)
-		return nil
+	if err := c.deleteServiceRoute(svcIP); err != nil {
+		return fmt.Errorf("failed to delete route for LoadBalancer ingress IP %s: %w", svcIPStr, err)
 	}
-	if err := c.netlink.RouteDel(route.(*netlink.Route)); err != nil {
-		if err.Error() == "no such process" {
-			klog.InfoS("Failed to delete route for LoadBalancer ingress IP since it doesn't exist", "route", route)
-		} else {
-			return fmt.Errorf("failed to delete route for LoadBalancer ingress IP %s: %w", svcIPStr, err)
-		}
+	klog.V(4).InfoS("Deleted route for LoadBalancer ingress IP", "IP", svcIPStr)
+	return nil
+}
+
+// AddExternalIP is used to add routing entry which is used to route ExternalIP to Antrea gateway on host.
+func (c *Client) AddExternalIP(svcIP net.IP) error {
+	svcIPStr := svcIP.String()
+	if err := c.addServiceRoute(svcIP); err != nil {
+		return fmt.Errorf("failed to install route for ExternalIP %s: %w", svcIPStr, err)
 	}
-	klog.V(4).InfoS("Deleted LoadBalancer ingress IP route", "route", route)
-	c.serviceRoutes.Delete(svcIPStr)
+	klog.V(4).InfoS("Added route for ExternalIP", "IP", svcIPStr)
+	return nil
+}
 
+// DeleteExternalIP is used to delete routing entry which is used to route ExternalIP to Antrea gateway on host.
```

Comment:
DeleteExternalIP deletes the route entry for a Service ExternalIP.

Reply:
Maybe you can consider updating the comments for Add/DeleteLoadBalancer too.

Reply:
Updated `Add/DeleteLoadBalancer` to `Add/DeleteLoadBalancerIngressIPRoute`, `Add/DeleteExternalIP` to `Add/DeleteExternalIPRoute`.